### ISSUE TITLE
Installing Discovery plugin on windows 10 2.18.14

### ISSUE THREAD

**Issue by joeyork** [lutraconsulting/qgis-discovery-plugin#22]
I have tried to install the plugin but I keep getting this message

Couldn't load plugin Discovery due to an error when calling its initGui() method 

TypeError: unable to convert a QVariant of type 0 to a QMetaType of type 10 

Traceback (most recent call last):

  File "C:/PROGRA~1/QGIS21~1.14/apps/qgis/./python\qgis\utils.py", line 342, in startPlugin

    plugins[packageName].initGui()

  File "C:/Users/cbc_admin/.qgis2/python/plugins\Discovery\discoveryplugin.py", line 205, in initGui

    self.read_config(config_list[0] if config_list else "")

  File "C:/Users/cbc_admin/.qgis2/python/plugins\Discovery\discoveryplugin.py", line 355, in read_config

    connection = settings.value(key + "connection", "", type=str)

TypeError: unable to convert a QVariant of type 0 to a QMetaType of type 10

Python version: 2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)] 

QGIS version: 2.18.14 Las Palmas, ff83b9a479 

Python Path:

•C:/PROGRA~1/QGIS21~1.14/apps/qgis/./python/plugins\processing

•C:/PROGRA~1/QGIS21~1.14/apps/qgis/./python

•C:/Users/cbc_admin/.qgis2/python

•C:/Users/cbc_admin/.qgis2/python/plugins

•C:/PROGRA~1/QGIS21~1.14/apps/qgis/./python/plugins

•C:\PROGRA~1\QGIS21~1.14\bin\python27.zip

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\DLLs

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\plat-win

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\lib-tk

•C:\PROGRA~1\QGIS21~1.14\bin

•C:\PROGRA~1\QGIS21~1.14\apps\Python27

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\jinja2-2.7.2-py2.7.egg

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\markupsafe-0.23-py2.7-win-amd64.egg

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\win32

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\win32\lib

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\Pythonwin

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\Shapely-1.2.18-py2.7-win-amd64.egg

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\wx-2.8-msw-unicode

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\xlrd-0.9.2-py2.7.egg

•C:\PROGRA~1\QGIS21~1.14\apps\Python27\lib\site-packages\xlwt-0.7.5-py2.7.egg

•C:/Users/cbc_admin/.qgis2//python

Cheers

Joe York

**Comment by sklencar** [comment]
Hi @joeyork,
thanks for the feedback. The bug is linked with loading config from QSettings which has been set in previous versions (<= 1.3). However, it should work now from version 1.4.1. Please, check it out.

Cheers,
Viktor Sklencar

**Comment by karl-magnus** [comment]
Hi!
I've kept getting this error on some clients even though I've uninstalled and reinstalled the plugin. The only way to fix it, that I could come up with, was to delete the Discovery entry in the registry. Is there still more to fix? Here is a screenshot of the invalid registry content.
Cheers
Karl-Magnus
![bild](https://user-images.githubusercontent.com/4749245/37965338-0c566f34-31c5-11e8-90c1-ea8eaa2bf727.png)
